static/app/views/alerts/rules/metric/details/body.tsx

```diff
@@ -236,6 +237,16 @@ export default function MetricDetailsBody({
             )}
           </StyledSubHeader>
 
+          {selectedIncident?.alertRule.detectionType ===
+            AlertRuleComparisonType.DYNAMIC && (
+            <AnomalyDetectionFeedbackBanner
+              key={selectedIncident.id}
```

Comment:
Comment to note why `key` is important here would be helpful for future visitors.

```suggestion
              // unique key to force re-render when incident changes
              key={selectedIncident.id}
```